crowdsourcing/wpconvlib/src/conversation.ts

```diff
@@ -170,14 +198,14 @@ export function htmlForComment(
 export function walkDfsComments(
     rootComment: Comment, f: (c: Comment) => void): void {
   const commentsHtml = [];
-  let agenda: Comment[] = [];
-  let nextComment: Comment|undefined = rootComment;
-  while (nextComment) {
+  const nextComment: Comment|undefined = rootComment;
+  if (nextComment) {
+    f(nextComment);
     if (nextComment.children) {
-      agenda = agenda.concat(nextComment.children);
+      for (const ch of nextComment.children) {
+        walkDfsComments(ch, f)
```

Comment:
This is doing DFS using the stack... that's probably ok, although usually in imperative languages you don't have a big stack space, but you have a big heap space; so usually people say you should write DFS algorithms for imperative languages using the heap if your size will get big. You can imply add comment here to clarify that this will only work for relatively shallow conversations, which is ok because we don't expect conversations to get more deeply nested than 1000.